### ISSUE TITLE
Problems signing in

### ISSUE THREAD

**Issue by joelyuk** [d8ahazard/Phlex#69]
Hi,

I've setup Phlex on a CentOS 7.3 server with PHP 7.0, with nginx and PHP-FPM.

When I try to sign in, I just get a response of 'ERROR' from api.php. After spending some time debugging, it seems to be the cURL request to Plex that is breaking, and I've added debug to try and capture where the curl request was failing:

```
string(1376) "* About to connect() to plex.tv port 443 (#0)
*   Trying 34.250.125.170...
* Connected to plex.tv (34.250.125.170) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /var/www/[removed]/html/cert/cacert.pem
  CApath: none
* SSL connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
* Server certificate:
* 	subject: CN=plex.tv,O="Plex, Inc",L=Los Gatos,ST=CA,C=US,postalCode=95030,STREET=449 N Santa Cruz Ave,serialNumber=4762731,incorporationState=Delaware,incorporationCountry=US,businessCategory=Private Organization
* 	start date: Oct 14 00:00:00 2015 GMT
* 	expire date: Jan 11 12:00:00 2018 GMT
* 	common name: plex.tv
* 	issuer: CN=DigiCert SHA2 Extended Validation Server CA,OU=www.digicert.com,O=DigiCert Inc,C=US
> POST /users/sign_in.xml HTTP/1.1
Host: plex.tv
Accept: */*
X-Plex-Client-Identifier: 4b6797b5606efb1fbd64760f
X-Plex-Device:PhlexWeb
X-Plex-Device-Screen-Resolution:1520x707,1680x1050,1920x1080
X-Plex-Device-Name:Phlex
X-Plex-Platform:Web
X-Plex-Platform-Version:1.0.0
X-Plex-Product:Phlex
X-Plex-Version:1.0.0
X-Plex-Provides:player,controller,sync-target,pubsub-player
Authorization:Basic [removed]
Content-Length: -1
Content-Type: application/x-www-form-urlencoded
Expect: 100-continue

< HTTP/1.1 400 BAD_REQUEST
< Content-Length: 0
< Connection: Close
< 
* Closing connection 0
"
```

Seems to be returning a 400 bad request but can't figure out why that is?

Thanks!

**Comment by farmjope** [comment]
I can't even get that far with the latest code..... The old code the UI comes up and everything, but with the latest code all I get is a black screen.

**Comment by d8ahazard** [comment]
Are you getting anything in Phlex_error.log that may be of use?

**Comment by d8ahazard** [comment]
@joelyuk  - Give today's commits a try.  I did some major reworking on a bunch of things that may have caused your issues.  If not, let me know what you see in Phlex_error.log and Phlex.log.

**Comment by mgiljum** [comment]
I get the followin in Phlex_error.log when trying to sign in:

```
#0 /var/www/html/Phlex/api.php(4095): signIn('bWF0aGV3LmdpbGp...')
#1 /var/www/html/Phlex/api.php(13): checkSignIn()
#2 {main}
  thrown in /var/www/html/Phlex/util.php on line 507
[04-Jun-2017 21:38:41 America/New_York] PHP Fatal error:  Uncaught Error: Class 'SimpleXMLElement' not found in /var/www/html/Phlex/util.php:507
Stack trace:
#0 /var/www/html/Phlex/api.php(4095): signIn('bWF0aGV3LmdpbGp...')
#1 /var/www/html/Phlex/api.php(13): checkSignIn()
#2 {main}
  thrown in /var/www/html/Phlex/util.php on line 507
```

and the following in Phlex.log:

```
Sun, 04 Jun 2017 21:38:41 -0400 [ DEBUG ] api.php::checkSignIn - Function fired.
Sun, 04 Jun 2017 21:38:41 -0400 [ DEBUG ] util.php::signIn - No token or not signed in, signing into Plex.
```

**Comment by d8ahazard** [comment]
@joelyuk @mgiljum - Is this still an issue for you guys?

**Comment by theone85ca** [comment]
Hey,

This is an issue for me. Clicking the 'Do It!' button seemingly does nothing. The API returns 200 with 'ERROR'. I've tried two separate servers, both my home one and a Vultr one.

Used wget to get the zip, unzipped and moved to the /var/www/html/Phlex and updated ownership to apache:apache. Installed PHP7.1 and restarted apache. CURL and OpenSSL modules are installed.

Theres nothing not already posted here in the logs.

Feel free to see the results here: http://45.76.19.139/Phlex/

**Comment by rets85** [comment]
@d8ahazard I'm having the exact same issue. No errors in console, the only thing I get in the log file is:

Fri, 07 Jul 2017 13:50:58 +0000 [ DEBUG ] api.php::checkSignIn - Function fired.
Fri, 07 Jul 2017 13:50:58 +0000 [ DEBUG ] util.php::signIn - No token or not signed in, signing into Plex.
Fri, 07 Jul 2017 13:51:04 +0000 [ DEBUG ] util.php::signIn - Container: null

When trying to fill in my login and password and clicking "Go for it!"

**Comment by joelyuk** [comment]
I'm still having the same issue - the login just returns 'ERROR' on the api.php response.

There is no Phlex.log or Phlex_error.log either in the root. I removed and re-cloned the whole project and it made no difference either.

**Comment by d8ahazard** [comment]
Does the account that the web server is running as have permissions to
write to the Phlex folder?

**Comment by joelyuk** [comment]
It does indeed, the folder (& files) are owned by the PHP FPM process so should be able to write fine if needed. Even tried chmod-ing to 777 just in case and still no signs of a log, and the same ERROR response from api.php

**Comment by d8ahazard** [comment]
Anything weird with open_basedir?

http://php.net/manual/en/ini.core.php#ini.open-basedir

**Comment by joelyuk** [comment]
Nope thats disabled too 😞

**Comment by d8ahazard** [comment]
See anything in the apache error logs?

**Comment by muyiwao221** [comment]
i'm having the same issue. Installed Phlex on Apache. Cant sign in. I put in my plex credentials and click "Do It" - then a sign comes up saying "One moment please" and it stays like that forever. No sign in. Any ideas whats up?

**Comment by jervine** [comment]
Trying to set this up/run in a CentOS 7.3 docker container, and have the same symptoms as described above - the /var/www/html/Phlex/logs/Phlex.log file contains the following entries when a login is attempted:
  [Sun, 23 Jul 2017 05:20:14 +0000] [INFO] [api.php::checkSignIn] - Trying to sign into Plex.tv as <_username_>.
  [Sun, 23 Jul 2017 05:20:14 +0000] [DEBUG] [util.php::signIn] - No token or not signed in, signing into Plex.

The Phlex_error.log is empty, and there are no entries in the /var/logs/httpd/error_log either.

**Comment by jervine** [comment]
As a test, I've tried Phlex on a Fedora 26 installation, and it has worked. Perhaps there is something strange with the remi PHP repo.

**Comment by d8ahazard** [comment]
@joelyuk - Still wondering if you're fighting this or have gotten some semblance of success?  Please let me know, or I'll mark this issue as closed in the next few days.

**Comment by rets85** [comment]
Still does not work for me. I believe @joelyuk and I have the same issue.

**Comment by pskipworth** [comment]
Logging same issue as above on CentOS 7 and XAMPP under OS X. Just tried it under Fedora 26, and I get further than before, but still having problems (will log new issue). Thanks for the work you've done on Phlex so far!

**Comment by Tin-Bui** [comment]
I got the same problem on Ubuntu 16 on my OdroidxU4 as well. 
Phlex.log:
[Sat, 28 Oct 2017 20:20:41 +1100] [INFO] [api.php::checkSignIn] - Trying to sign into Plex.tv as <username>.
[Sat, 28 Oct 2017 20:20:41 +1100] [DEBUG] [util.php::signIn] - No token or not signed in, signing into Plex.
----------------------------
I also saw on apache log errors to load the 2 additional extensions included in the installation guide:
extension=curl.so
extension=openssl.so

Turn out they are already enabled (I think, I am very novice in Linux):
--------------------
phpinfo()
...
cURL support | enabled
cURL Information | 7.47.0
...
OpenSSL support | enabled
OpenSSL Library Version | OpenSSL 1.0.2g 1 Mar 2016
OpenSSL Header Version | OpenSSL 1.0.2g 1 Mar 2016
---------------------

So I disabled them. Restart apache and no more error in the apache log.
However, I didn't get any further with the Phlex login page.
Still the same error and error message in the Phlex.log.
